crates/holochain_sqlite/src/sql/p2p_agent_store/gossip_query.sql

```diff
@@ -2,33 +2,28 @@ SELECT agent
 FROM p2p_agent_store
 WHERE signed_at_ms >= :since_ms -- between given signed_at range
   AND signed_at_ms <= :until_ms
+  -- if a null range is passed in, return no result
+  -- TODO: is this check actually necessary? if both are null,
+  --       both cases will return false anyway.
+  AND :storage_start_loc IS NOT NULL
+  AND :storage_end_loc IS NOT NULL
   AND (
-    -- if the input has two ranges, check them both
     (
-      :storage_start_1 IS NOT NULL
-      AND :storage_end_1 IS NOT NULL
-      AND :storage_start_2 IS NOT NULL
-      AND :storage_end_2 IS NOT NULL
+      -- non-wrapping case: everything within the given range
+      :storage_start_loc <= :storage_end_loc
       AND (
-        (
-          storage_center_loc >= :storage_start_1
-          AND storage_center_loc <= :storage_end_1
-        )
-        OR
-        (
-          storage_center_loc >= :storage_start_2
-          AND storage_center_loc <= :storage_end_2
-        )
+          storage_center_loc >= :storage_start_loc
+          AND storage_center_loc <= :storage_end_loc
       )
     )
     OR
-    -- if the input has only one range, check it
     (
-      :storage_start_1 IS NOT NULL
-      AND :storage_end_1 IS NOT NULL
-      AND storage_center_loc >= :storage_start_1
-      AND storage_center_loc <= :storage_end_1
+      -- wrapping case: everything *outside* the given range
+      :storage_start_loc > :storage_end_loc
+      AND (
+          storage_center_loc < :storage_end_loc
+          OR storage_center_loc > :storage_start_loc
```

Comment:
Even though it wraps around, the marked points are still inclusive, so this needs to be <= / >= no?

```suggestion
          storage_center_loc <= :storage_end_loc
          OR storage_center_loc >= :storage_start_loc
```